amp/WEB-INF/src/org/digijava/module/aim/util/ActivityVersionUtil.java

```diff
@@ -6,325 +6,696 @@
+import org.digijava.module.aim.helper.ActivityHistory;
 import org.digijava.module.aim.helper.DateConversion;
 import org.digijava.module.aim.helper.FormatHelper;
 import org.digijava.module.aim.helper.GlobalSettingsConstants;
 import org.digijava.module.aim.version.exception.CannotGetLastVersionForVersionException;
+import org.hibernate.Hibernate;
 import org.hibernate.Query;
 import org.hibernate.Session;
 
 public class ActivityVersionUtil {
 
-    private static Logger logger = Logger.getLogger(ActivityVersionUtil.class);
+	private static Logger logger = Logger.getLogger(ActivityVersionUtil.class);
 
+	public static Method getMethodFromFieldName(String fieldName, Class auxClass, String prefix) throws Exception {
+		String methodName = fieldName.substring(0, 1).toUpperCase() + fieldName.substring(1);
+		methodName = prefix + methodName;
+		if (prefix.equals("set")) {
+			for (int i = 0; i < auxClass.getDeclaredMethods().length; i++) {
+				if (auxClass.getDeclaredMethods()[i].toString().contains(methodName)) {
+					return auxClass.getDeclaredMethods()[i];
+				}
+			}
+		}
+		try {
+			return auxClass.getMethod(methodName, null);
+		} catch (Exception e) {
+			if (prefix.equals("get")) {
+				// boolean fields might have "is" as a prefix
+				return getMethodFromFieldName(fieldName, auxClass, "is");
+			}
+			throw e;
+		}
+	}
 
-    public static Method getMethodFromFieldName(String fieldName, Class auxClass, String prefix) throws Exception {
-        String methodName = fieldName.substring(0, 1).toUpperCase() + fieldName.substring(1);
-        methodName = prefix + methodName;
-        if (prefix.equals("set")) {
-            for (int i = 0; i < auxClass.getDeclaredMethods().length; i++) {
-                if (auxClass.getDeclaredMethods()[i].toString().contains(methodName)) {
-                    return auxClass.getDeclaredMethods()[i];
-                }
-            }
-        }
-        try {
-            return auxClass.getMethod(methodName, null);
-        }
-        catch(Exception e) {
-            if (prefix.equals("get")) {
-                // boolean fields might have "is" as a prefix
-                return getMethodFromFieldName(fieldName, auxClass, "is");
-            }
-            throw e;
-        }
-    }
+	public static boolean implementsVersionable(Class[] array) {
+		boolean ret = false;
+		for (int i = 0; i < array.length; i++) {
+			if (array[i].getName().equals("org.digijava.module.aim.dbentity.Versionable")) {
+				ret = true;
+			}
+		}
+		return ret;
+	}
 
-    public static boolean implementsVersionable(Class[] array) {
-        boolean ret = false;
-        for (int i = 0; i < array.length; i++) {
-            if (array[i].getName().equals("org.digijava.module.aim.dbentity.Versionable")) {
-                ret = true;
-            }
-        }
-        return ret;
-    }
+//	public static String generateFormattedOutput(HttpServletRequest request, Output out) throws WorkerException {
```

Comment:
as I said in other class we should try to avoid commenting code and just changing what's need to be changed. If you see the pr changes its very difficult to trace what has changed. Its usually difficult for the developer that is reviewing the code tell what has changed